### PR TITLE
Set rustdoc --test files' path relative to the current directory

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @alexcrichton

### CODE REVIEW

File: src/librustdoc/test.rs [hunk 1]
```diff
@@ -485,7 +485,15 @@ impl Collector {
 
     pub fn get_filename(&self) -> String {
         if let Some(ref codemap) = self.codemap {
-            codemap.span_to_filename(self.position)
+            let mut filename = codemap.span_to_filename(self.position);
+            if let Ok(cur_dir) = env::current_dir() {
+                if let Some(cur_dir) = cur_dir.to_str() {
```
Comment (alexcrichton) [inline src/librustdoc/test.rs]: This should use `Path::strip_prefix` if possible

Comment (GuillaumeGomez) [inline src/librustdoc/test.rs]: Taking a look!

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by GuillaumeGomez** [comment]
@alexcrichton: On appveyor, the tests fail because the path look like this: `src/abc\def\ghi\jkl`. Why is there a `/` on windows? O.o

**Comment by alexcrichton** [comment]
@GuillaumeGomez yes, that's also a standard path separator on Windows

**Comment by GuillaumeGomez** [comment]
Sure, but it's a bit strange to have only one `\` whereas there are only `/`. And I have no clue how to enforce the first or the other one to avoid test failure...

**Comment by GuillaumeGomez** [comment]
@alexcrichton: So I forced paths to only use `/` in the compiletest. Does it seem ok to you?

**Comment by GuillaumeGomez** [comment]
cc @alexcrichton

**Comment by GuillaumeGomez** [comment]
Ok, I fixed the issue. I'll r+ it but don't hesitate to r- if you see anything @alexcrichton.

@bors: r=alexcrichton

**Comment by alexcrichton** [comment]
@bors: retry

* network error

**Comment by alexcrichton** [comment]
@bors: retry

* empty logs?

**Comment by eddyb** [comment]
@bors retry
 * manually cancelled to prioritize rollup

**Comment by eddyb** [comment]
@bors retry
 * manually cancelled to prioritize rollup

**Pull request merged**